packages/expo-router/src/headless/Tabs.slot.tsx

```diff
@@ -0,0 +1,100 @@
+import { Slot } from '@radix-ui/react-slot';
+import { ComponentType, useState, useContext } from 'react';
+import { View, Platform, ViewProps, StyleSheet } from 'react-native';
+import { ScreenContainer, Screen } from 'react-native-screens';
+import { TabsDescriptor, TabsDescriptorsContext, TabsStateContext } from './Tabs.common';
+
+export type UseTabSlotOptions = {
+  detachInactiveScreens?: boolean;
+  renderFn?: typeof defaultTabsSlotRender;
+};
+
+export type TabsSlotRenderOptions = {
+  index: number;
+  isFocused: boolean;
+  loaded: boolean;
+};
+
+export function useTabSlot({
+  detachInactiveScreens = Platform.OS === 'web' ||
```

Comment:
This matches the logic in React Navigation. I'm not sure why the other out of tree platfoms are not allowed.